### PR TITLE
MGDOBR-699:[UI] add test for checking instance header details

### PR DESCRIPTION
**Pull request by sthundat**
Jira:[https://issues.redhat.com/browse/MGDOBR-699](https://issues.redhat.com/browse/MGDOBR-699)

### CODE REVIEW

File: cypress/integration/smoke/BasicTest.ts [hunk 1]
```diff
@@ -39,4 +39,21 @@ describe("Basic Elements", () => {
...
+      "Time created",
+    ];
+    cy.get(
+      "table[data-ouia-component-id='Instances list table'] thead tr th"
+    ).should("have.length", 4);
+    cy.get(
+      "table[data-ouia-component-id='Instances list table'] thead tr th"
+    ).each((item, index) => {
+      cy.wrap(item).should("contain.text", instanceHeaderDetails[index]);
+    });
```
Comment (jomarko) [inline cypress/integration/smoke/BasicTest.ts:57]: @sjamboro I believe this can be rewritten somehow with your `ouia` related PRs?

**Review by jomarko — COMMENTED** [review]
One comment related to how we match elements using `ouia` based locators. Lets wait what @sjamboro replies.

File: cypress/integration/smoke/BasicTest.ts [hunk 2]
```diff
@@ -39,4 +39,21 @@ describe("Basic Elements", () => {
...
   });
+
+  it("Instance header details are available", () => {
+    const instanceHeaderDetails = [
+      "Name",
+      "Description",
+      "Status",
+      "Time created",
+    ];
+    cy.get(
```
Comment (sjamboro) [inline cypress/integration/smoke/BasicTest.ts:50]: You can use the `ouiaId` command. See https://github.com/sjamboro/sandbox-ui/blob/MGDOBR-654/cypress/integration/smoke/BasicTest.ts#L33-L39

File: cypress/integration/smoke/BasicTest.ts [hunk 3]
```diff
@@ -39,4 +39,21 @@ describe("Basic Elements", () => {
...
+      "table[data-ouia-component-id='Instances list table'] thead tr th"
+    ).should("have.length", 4);
+    cy.get(
```
Comment (sjamboro) [inline cypress/integration/smoke/BasicTest.ts:53]: I think that you should not repeat the searching for the same element. See: https://github.com/sjamboro/sandbox-ui/blob/MGDOBR-654/cypress/integration/InstancesTest.ts#L111-L124

File: cypress/integration/smoke/BasicTest.ts [hunk 4]
```diff
@@ -39,4 +39,21 @@ describe("Basic Elements", () => {
...
+      "table[data-ouia-component-id='Instances list table'] thead tr th"
+    ).each((item, index) => {
```
Comment (sjamboro) [inline cypress/integration/smoke/BasicTest.ts:55]: I appreciate this line. It is more readable and better maintaining solution than my. See https://github.com/sjamboro/sandbox-ui/blob/MGDOBR-654/cypress/integration/InstancesTest.ts#L111-L124.

File: cypress/integration/smoke/BasicTest.ts [hunk 5]
```diff
@@ -39,4 +39,21 @@ describe("Basic Elements", () => {
...
+      cy.wrap(item).should("contain.text", instanceHeaderDetails[index]);
```
Comment (sjamboro) [inline cypress/integration/smoke/BasicTest.ts:56]: Is there any problem with "have.text"?

File: cypress/integration/smoke/BasicTest.ts [hunk 6]
```diff
@@ -39,4 +39,21 @@ describe("Basic Elements", () => {
...
+  it("Instance header details are available", () => {
+    const instanceHeaderDetails = [
+      "Name",
+      "Description",
+      "Status",
+      "Time created",
+    ];
+    cy.get(
+      "table[data-ouia-component-id='Instances list table'] thead tr th"
+    ).should("have.length", 4);
```
Comment (sjamboro) [inline cypress/integration/smoke/BasicTest.ts:52]: Do you think about `instanceHeaderDetails.length` instead of `4`?

**Review by sjamboro — CHANGES_REQUESTED** [review]

File: cypress/integration/smoke/BasicTest.ts (hunk 3 above)
Comment (sthundat) [reply cypress/integration/smoke/BasicTest.ts]: done

File: cypress/integration/smoke/BasicTest.ts (hunk 5 above)
Comment (sthundat) [reply cypress/integration/smoke/BasicTest.ts]: no problem!

File: cypress/integration/smoke/BasicTest.ts [hunk 7]
```diff
@@ -39,4 +39,21 @@ describe("Basic Elements", () => {
...
+
+  it("Instance header details are available", () => {
+    const instanceHeaderDetails = [
+      "Name",
+      "Description",
+      "Status",
+      "Time created",
+    ];
+    cy.get(
+      "table[data-ouia-component-id='Instances list table'] thead tr th"
```
[the remark this answers is not in the document]
Comment (sthundat) [reply cypress/integration/smoke/BasicTest.ts]: changed to `cy.ouiaId("Instances list table", "PF4/Table") .ouiaType("PF4/TableRow").find("th")`  please let me know this is fine

**Review by sjamboro — APPROVED** [review]
Thank you, LGTM

**Comment by sjamboro** [comment]
@jomarko I have take a look at failing test. I do not know why the `InstancesTest.ts` test runs here. It does not exist in this branch:
https://github.com/sthundat/sandbox-ui/tree/instanceheadertest/cypress/integration

The history of commits does not contain PR #38. See https://github.com/sthundat/sandbox-ui/commits/instanceheadertest

File: cypress/integration/smoke/BasicTest.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (sjamboro) [reply cypress/integration/smoke/BasicTest.ts]: @sthundat I have not got any problem with that. There is the only one TableRow.

**Review by jomarko — APPROVED** [review]
Approving, however would be good, to rebase PR with main, then force push, and see if the CI is green.

**Comment by sthundat** [comment]
> Approving, however would be good, to rebase PR with main, then force push, and see if the CI is green.

@jomarko Rebased and all checks passed!

**Pull request merged**